confignode/src/main/java/org/apache/iotdb/confignode/client/async/datanode/AsyncDataNodeClientPool.java

```diff
@@ -168,60 +190,86 @@ public void sendAsyncRequestToDataNode(
   public void createRegions(
       CreateRegionGroupsPlan createRegionGroupsPlan, Map<String, Long> ttlMap) {
 
-    // Number of regions to be created
-    int regionNum = 0;
-    // Assign an independent index to each Region
+    List<TDataNodeLocation> dataNodeLocations = new ArrayList<>();
+    // Count the datanodes to be sent
     for (Map.Entry<String, List<TRegionReplicaSet>> entry :
         createRegionGroupsPlan.getRegionGroupMap().entrySet()) {
       for (TRegionReplicaSet regionReplicaSet : entry.getValue()) {
-        regionNum += regionReplicaSet.getDataNodeLocationsSize();
+        regionReplicaSet
+            .getDataNodeLocations()
```

Comment:
can be replaced with `dataNodeLocations.addAll(regionReplicaSet.getDataNodeLocations());`

Reply:
fixed~